### PR TITLE
Updating approve/stake/withdraw UX

### PR DESCRIPTION
**Pull request by audieleon**
### Made Buttons on MyWallet operate independently and on their own criteria

- Made UX get and use allowed tokens
   - Added allowed tokens to actions
- Made buttons operate independently
- Change the Text above the balance box based on the chosen action (it's not always Stake...)

> TODO _Still need to correctly set button state and styles for this..._

@JohannesHa and/or @schmackofant Could you review the approach here and give me feedback? 

I'm trying to make it so that the buttons on this UI that should work _do work_, and do so independently of the other button's state. For example: 

- I can and should always be able to approve tokens, regardless of other button state. 
- I can only stake if I have approved tokens, and if the stake amount is less than my approved tokens minus my already staked tokens. 
- I can only withdraw if I have staked tokens and the withdraw date is either today or in the past. 

I've not implemented the above states yet, and I'll need to restyle the PillButtons based on the state (green button if it should work, dead and grey if it won't - no highlight and no cursor change). I also want to add tooltips to explain a few things, and set the spinner to the right statuses once clicked (then have the spinner clear after a short time). 

Trying to make the UX for this make sense. 

**Again - looking for feedback on the approach. Do you see a reason this wouldn't work or a way to improve it?**

### CODE REVIEW

**Comment by schmackofant** [comment]
I really like the idea, that these buttons and approve/stake/withdraw can work independently 👍 . Clearly indicating to the user which actions are possible right now would be great as well. 

* We probably shouldn't rely too much on mouse events for the final UX, though. So that we don't exclude touch devices. Tooltips can ideally get triggered by touch action as well (maybe with a corresponding help icon or similar).  
* Not sure if the number of "approved tokens" should even be displayed. Normally as a user, I shouldn't have to care and it might be confusing to some. We could display a warning contextually, if I try to stake more than I have approved. Hoping for https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2612.md on that front anyway ;).

**Comment by audieleon** [comment]
Made the three buttons on the My Wallet Page react to the only the conditions that affect it. 

Wallet States For Testing (the ones I could think of): 

- Initial State (Wallet Disconnected): 
  - Amount: 0 | Staked: blank  | Balance: 0 | Allowed: 0 | UnlockDate: blank
    - One grey button that says "please connect wallet"
- Initial State (Connected Wallet): 
  - Amount: 0 | Staked: 0 | Balance: 100 | Allowed: 0 | UnlockDate: Past
     - All Three Buttons should be grey and read: "Enter Approval Amount", "Cannot Stake Yet", "Cannot Withdraw Yet"
     - Clicking any button does nothing 
- Ready To Approve: 
  - Amount: 100 | Staked: 0 | Balance: 100 | Allowed: 0 | UnlockDate: Past
    - Approve Token turns green and will initiate transaction in wallet
- Trying To Approve Too Much: 
 -  Amount: 1000 | Staked: 0 | Balance: 100 | Allowed: 0 | UnlockDate: Past
    - Buttons drop to one grey button that reads "Insufficient Vita Balance"    
- Approved
  - Amount: 0 | Staked: 0 | Balance: 100 | Allowed: 100 | UnlockDate: Past
    - Stake Button still grey due to zero balance
- Ready To Stake
  - Amount: 10 | Staked: 0 | Balance: 100 | Allowed: 100 | UnlockDate: Past
    - Stake button turns green and will initiate transaction in wallet
- Staked
  - Amount: 0 | Staked: 10 | Balance: 100 | Allowed: 90 | UnlockDate: Past
    - First two buttons go back grey, withdraw button highlights and will initiate transaction in wallet
- Ready To Withdraw: 
  - Amount: 10 | Staked: 10 | Balance: 100 | Allowed: 90 | UnlockDate: Past
    - First Button Green and Reads "Approve Tokens"
    - Second Button grey and reads: "Cannot Stake Yet"
    - Withdraw Green and will initiate withdraw transaction in wallet
- Not Ready To Withdraw: 
  - Amount: 0 | Staked: 10 | Balance: 100 | Allowed: 100 | UnlockDate: Future (need to vote)

I checked all of these, and they seemed to work without issue.

**Comment by schmackofant** [comment]
I noticed an irregularity with approval. I tried to approve 40 tokens but the approval TX created was for 41 tokens. It's always 1 more than I entered (calc error?).

Also "cannot stake yet" never changes, it seems. I approved multiple times but couldn't stake those tokens. Refresh and reconnect also didn't help.

Bildschirmfoto 2021-07-22 um 14 30 26

File: src/store/services/token.ts [hunk 1]
```diff
@@ -13,12 +13,12 @@ export const approveTokens = async (payload: any)=>{
   try {
     const { contracts, address, amount} = payload;
     const { tokenContract, stakingContract } = contracts;
-    let changedAmount = amount + 1;
+    let changedAmount = parseInt(amount) + 1;
```
Comment (schmackofant) [inline src/store/services/token.ts:16]: this seems to be responsible for the token approval always being 1 more than entered

**Comment by audieleon** [comment]
> I noticed an irregularity with approval. I tried to approve 40 tokens but the approval TX created was for 41 tokens. It's always 1 more than I entered (calc error?).
> 
> Also "cannot stake yet" never changes, it seems. I approved multiple times but couldn't stake those tokens. Refresh and reconnect also didn't help.


This was a mistake of mine. I assumed the approved tokens didn't change when you staked. I removed this incorrect assumption and you should now be able to take any amount of tokens above zero.

**Review by schmackofant — APPROVED** [review]

**Pull request merged**